Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -10,299 +10,400 @@
+        Block NextBlock(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Adds a specific requirement to all peer nodes.
+        /// </summary>
+        /// <param name="transactionOptions">Specifies the requirement on nodes to add.</param>
+        void RequestOptions(TransactionOptions transactionOptions);
+    }
+
+    /// <summary>
+    /// Puller that is used for fast sync during initial block download (IBD). It implements a strategy of downloading 
+    /// multiple blocks from multiple peers at once, so that IBD is fast enough, but does not consume too many resources.
+    /// </summary>
+    /// <remarks>
+    /// The node is aware of the longest chain of block headers, which is stored in this.Chain. This is the chain the puller 
+    /// needs to download. Here is a visualization of the block chain and how the puller sees it:
+    /// <para>
+    /// -------A------B-------------C-----------D---------E--------
+    /// </para>
+    /// <para>
+    /// Each '-' represents a block and letters 'A' to 'E' represent blocks with important positions in the chain from 
+    /// the puller's perspective. The puller can be understood as a producer of the blocks (by requesting them from the peers 
+    /// and downloading them) for the component that uses the puller that consumes the blocks (e.g. validating them).
+    /// </para>
+    /// <para>
+    /// A is a position of a block that we call location. Blocks in front of A were already downloaded and consumed.
+    /// </para>
+    /// <para>
+    /// B is a position of a block A + MinimumLookahead, and E is a position of block A + MaximumLookahead. 
+    /// Blocks between B and E are the blocks that the puller is currently interested in. Blocks after E are currently 
+    /// not considered and will only be interesting later. The lower boundary B prevents the IBD to be too slow, 
+    /// while the upper boundary E prevents the puller from using too many resources.
+    /// </para>
+    /// <para>
+    /// Blocks between A and B are blocks that have been downloaded already, but the consumer did not consume them yet.
+    /// </para>
+    /// <para>
+    /// C is a position of a block A + locationLookahead. Blocks between B and C are currently being requested by the puller, 
+    /// some of them could be already being downloaded.
+    /// </para>
+    /// <para>
+    /// D is a position of a block A + ActualLookahead. ActualLookahead is a number of blocks that the puller wants 
+    /// to be downloading simultaneously. If there is a gap between C and D it means that the puller wants to start 
+    /// downloading these blocks.
+    /// </para>
+    /// <para>
+    /// Blocks between D and E are currently those that the puller does not want to be downloading right now, 
+    /// but should the ActualLookahead be adjusted, they can be requested in the near future.
+    /// </para>
+    /// </remarks>
```

Comment:
It may seem that my explanation is complex, but it comes with a lot of additional details that your version is missing. I think those can be actually useful to anyone coming to the code later. 

I would thus suggest to modify the explanation in a way to expand it even more - to use what you wrote here, but not to lose the additional information. I will make another commit that will try to do that.

But in general, I would like to see more rather then less.